File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/repository/ChatParticipantRepository.java
package io.github.hs96wings.streaming_server.chat.repository;

import io.github.hs96wings.streaming_server.chat.domain.ChatParticipant;
import io.github.hs96wings.streaming_server.chat.domain.ChatRoom;
import io.github.hs96wings.streaming_server.member.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ChatParticipantRepository extends JpaRepository<ChatParticipant, Long> {
    List<ChatParticipant> findByChatRoom(ChatRoom chatRoom);
    Optional<ChatParticipant> findByChatRoomAndMember(ChatRoom chatRoom, Member member);
}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/repository/ChatParticipantRepository.kt
package io.github.hs96wings.streaming_server.chat.repository

import io.github.hs96wings.streaming_server.chat.domain.ChatParticipant
import io.github.hs96wings.streaming_server.chat.domain.ChatRoom
import io.github.hs96wings.streaming_server.member.domain.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ChatParticipantRepository : JpaRepository<ChatParticipant, Long> {
    fun findByChatRoom(chatRoom: ChatRoom): List<ChatParticipant>
    fun findByChatRoomAndMember(chatRoom: ChatRoom, member: Member): Optional<ChatParticipant>
    fun findAllByMember(member: Member): List<ChatParticipant>
}
